pahfit/model.py

```diff
@@ -734,158 +758,169 @@ def tabulate(
 
-        # h2_info
-        param_info[2] = PAHFITBase.update_dictionary(
-            param_info[2],
-            instrumentname,
-            update_fwhms=use_instrument_fwhm,
-            redshift=redshift,
-        )
-        # ion_info
-        param_info[3] = PAHFITBase.update_dictionary(
-            param_info[3],
-            instrumentname,
-            update_fwhms=use_instrument_fwhm,
-            redshift=redshift,
-        )
-        # abs_info
-        param_info[4] = PAHFITBase.update_dictionary(
-            param_info[4], instrumentname, redshift
-        )
+         instrumentname : str
+            Qualified instrument name
 
-        return param_info
+         x : array
+            Optional observed wavelength grid. Exclude any lines and
+            dust features outside of this range.
 
-    def _construct_astropy_model(
-        self, instrumentname, redshift, use_instrument_fwhm=True
-    ):
-        """Convert the features table into a fittable model.
+        Returns
+        -------
+        array of bool, same length as self.features, True where features
+        are far outside the wavelength range.
+        """
+        observed_wavs = self.features["wavelength"]["val"] * (1 + redshift)
 
-        Some nuances in the behavior
-        - If a line has a fwhm set, it will be ignored, and replaced by
-          the calculated fwhm provided by the instrument model.
-        - If a line has been masked by _parse_astropy_result, and this
-          function is called again, those masks will be ignored, as the
-          data range might have changed.
+        # has wavelength and not within instrument range
+        is_outside = ~instrument.within_segment(observed_wavs, instrumentname)
 
-        TODO: Make sure the features outside of the data range are
-        removed. The instrument-based feature check is done in
-        _kludge_param_info(), but the observational data might only
-        cover a part of the instrument range.
+        # also apply observed range if provided
+        if x is not None:
+            is_outside |= (observed_wavs < np.amin(x)) | (observed_wavs > np.amax(x))
 
-        """
-        param_info = self._kludge_param_info(
-            instrumentname, redshift, use_instrument_fwhm
+        # restriction on the kind of feature that can be excluded
+        excludable = ["line", "dust_feature", "absorption"]
+        is_excludable = np.logical_or.reduce(
+            [kind == self.features["kind"] for kind in excludable]
         )
-        return PAHFITBase.model_from_param_info(param_info)
 
-    def _parse_astropy_result(self, astropy_model):
-        """Store the result of the astropy fit into the features table.
+        return is_outside & is_excludable
 
-        Every relevant value inside the astropy model, is written to the
-        right position in the features table.
+    def _set_up_fitter(
+        self, instrumentname, redshift, x=None, use_instrument_fwhm=True
```

Comment:
What is `x`?  Not documented.

Reply:
Changed to `lam` and added more documentation.